File: examples/bun/.harmont/pipeline.ts
import { pipeline, push, type PipelineDefinition } from "harmont";
import { js } from "harmont/toolchains";

const project = js.project({ runtime: "bun" });

const pipelines: PipelineDefinition[] = [
  {
    slug: "ci",
    triggers: [push({ branch: "main" })],
    pipeline: pipeline(project.run("build"), project.test(), project.run("lint"), {
      env: { CI: "true" },
      defaultImage: "ubuntu:24.04",
    }),
  },
];

export default pipelines;

File: examples/bun/.hm/pipeline.py
"""Bun example pipeline."""
from __future__ import annotations

import harmont as hm


@hm.pipeline(
    "ci",
    env={"CI": "true"},
    triggers=[hm.push(branch="main")],
)
def ci() -> tuple[hm.Step, ...]:
    project = hm.js.project(path=".", runtime="bun")
    return (
        project.run("build"),
        project.run("test"),
        project.run("lint"),
    )
